providers/test/Test.fst

```diff
@@ -296,31 +296,46 @@ let rec test_aead_loop (LB len vs) =
 let test_aead () : St unit =
   test_aead_loop aead_vectors_low
 
-let rec test_chacha20poly1305_loop (i: U32.t): St unit =
-  let open Test.Vectors.Chacha20Poly1305 in
-  if i `U32.gte` vectors_len then
-    ()
+#push-options "--z3rlimit 32"
+
+let test_chacha20poly1305_one (v: Test.Vectors.Chacha20Poly1305.vector): Stack unit (fun _ -> True) (fun _ _ _ -> True) =
+  let Test.Vectors.Chacha20Poly1305.Vector cipher_and_tag cipher_and_tag_len plain plain_len aad aad_len nonce nonce_len key key_len = v in
+  if not (
+    key_len = 32ul &&
+    nonce_len = 12ul &&
+    0ul `U32.lt` plain_len &&
+    (4294967295ul `U32.sub` 16ul) `U32.gte` plain_len &&
+    (plain_len `U32.div` 64ul) `U32.lte` (4294967295ul `U32.sub` aad_len) &&
+    cipher_and_tag_len = plain_len `U32.add` 16ul
+  ) then
+    C.String.print !$"Warning: skipping a chacha20poly1305 instance because bounds do not hold"
```

Comment:
should we make these fatal errors? if a test is poorly crafted then surely there's a big problem, isn't there?